components/constellation/constellation.rs

```diff
@@ -2125,7 +2186,8 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             let evicted_id = frame.prev.len()
                 .checked_sub(PREFS.get("session-history.max-length").as_u64().unwrap_or(20) as usize)
                 .and_then(|index| frame.prev.get_mut(index))
-                .and_then(|entry| entry.pipeline_id.take());
+                .and_then(|entry| entry.pipeline_id);
+            let evicted_id = if Some(frame.pipeline_id) == evicted_id { None } else { evicted_id };
```

Comment:
This will evict, for example, a pipeline where there are many entries after it with the same pipeline id. Should we not look to see if `frame.prev[N]` and `frame.prev[N+1]` have different pipeline ids?